uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/MediaPopupElementViewController.cpp

```diff
@@ -48,9 +55,40 @@ MediaPopupElementViewController::MediaPopupElementViewController(
   : PopupElementViewItem{mediaPopupElement, parent},
     m_popupMediaItems{new GenericListModel(&PopupMediaItem::staticMetaObject, this)}
 {
-  for (auto* media: *static_cast<MediaPopupElement*>(popupElement())->media())
+  auto* media = mediaPopupElement->media();
+  for (int i = 0; i < media->size(); i++)
   {
-    m_popupMediaItems->append(new PopupMediaItem(media, this));
+    const QJsonDocument mediaPopupElementJson = QJsonDocument::fromJson(mediaPopupElement->toJson().toUtf8());
+    auto* popupMedia = media->at(i);
+
+    // Get color information for each popup media. If no color is selected when defining a popup, the default
+    // color will be used from the appropriate QtGraphs Series component.
+    // s.a. Web Map Spec: https://developers.arcgis.com/web-map-specification/objects/popupElement_media
+    const auto colors = mediaPopupElementJson.object()["mediaInfos"][i]["value"]["colors"].toArray();
```

Comment:
I would make a helper for this that validates everything. This makes assumptions that will crash if they don't hold.